mlflow/models/dependencies_schemas.py

```diff
@@ -82,6 +86,17 @@ def set_retriever_schema(
                 name="my_custom_retriever",
             )
     """
+    warnings.warn(
+        "set_retriever_schema is deprecated and will be removed in a future version. "
+        "Please migrate to use VectorSearchRetrieverTool in the 'databricks-ai-bridge' package, "
+        "or match the default schema so your retriever spans can be detected without requiring "
+        "explicit configuration. See "
+        "https://mlflow.org/docs/latest/genai/data-model/traces/#retriever-spans "
+        "for more information.",
+        category=DeprecationWarning,
```

Comment:
By default, python doesn't show `DeprecationWarning`. `FutureWarning` should be used instead.